src/main/scala/com/itechart/ny_accidents/parse/AccidentsParser.scala

```diff
@@ -2,38 +2,39 @@ package com.itechart.ny_accidents.parse
 
 import java.time.LocalDateTime
 
-import com.google.inject.Singleton
 import com.itechart.ny_accidents.constants.AccidentsHeader._
 import com.itechart.ny_accidents.constants.GeneralConstants
-import com.itechart.ny_accidents.entity.Accident
+import com.itechart.ny_accidents.constants.GeneralConstants.ZERO_DATE
+import com.itechart.ny_accidents.entity.AccidentSparkFormat
 import com.itechart.ny_accidents.spark.Spark
+import com.itechart.ny_accidents.spark.Spark.sparkSql.implicits._
 import com.itechart.ny_accidents.utils.DateUtils
-import org.apache.spark.rdd.RDD
-import org.apache.spark.sql.{DataFrame, RelationalGroupedDataset, Row}
-import org.slf4j.LoggerFactory
+import org.apache.spark.sql._
 import org.apache.spark.sql.functions._
 import org.apache.spark.sql.types.StructType
+import org.slf4j.LoggerFactory
 
+import java.sql.{Date => SqlDate}
 import scala.util.Try
 
-@Singleton
-class AccidentsParser {
+
+object AccidentsParser {
```

Comment:
What are the benefits of replacing Singleton class with Object?

Reply:
It's out-of-box solution without 3party libs